contracts/wawaka/common/StringArray.cpp

```diff
@@ -185,6 +183,15 @@ bool StringArray::equal(const StringArray& sarray) const
     return (memcmp(value_, sarray.value_, size_) == 0);
 }
 
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+bool StringArray::null_terminated(void) const
+{
+    if (value_ == NULL || size_ == 0)
+        return false;
+
+    return (value_[size_ - 1] == 0);
```

Comment:
Since a string array is built from a binary buffer,
`{'h', 'i', '\0', 'M', 'i', 'c'}` is a valid binary buffer and string, but it's not null-terminated according to the definition/construction.
Is this intended?

Reply:
yes.